src/main/kotlin/io/sip3/salto/ce/media/MediaHandler.kt

```diff
@@ -124,20 +124,21 @@ open class MediaHandler : AbstractVerticle() {
         }
     }
 
-    open fun handleSdp(sessions: Pair<SdpSession, SdpSession>) {
-        val (request, response) = sessions
+    open fun handleMediaControl(mediaControl: MediaControl) {
+        val sdpSession = mediaControl.sdpSession
+        val srcAddr = sdpSession.src.rtpAddress()
+        val dstAddr = sdpSession.dst.rtpAddress()
 
-        val srcAddr = request.rtpAddress()
-        val dstAddr = response.rtpAddress()
-
-        media.getOrPut(request.callId) { mutableMapOf() }
-            .putIfAbsent(dstAddr.compositeAddrKey(srcAddr), MediaSession(srcAddr, dstAddr, request.callId))
+        media.getOrPut(mediaControl.callId) { mutableMapOf() }
+            .putIfAbsent(dstAddr.compositeAddrKey(srcAddr), MediaSession(srcAddr, dstAddr, mediaControl.callId).apply {
+                hasRecording = mediaControl.recording != null
```

Comment:
I disagree with this assignment. Because at this point we simple don't know if `sip3-captain` has managed to find and record the session.